packages/authorization-updater/src/authorizationService.ts

```diff
@@ -142,5 +160,39 @@ export const authorizationServiceBuilder = (
         `Updated Agreement ${agreementId} and EService ${eserviceId} states for all clients`
       );
     },
+    async deletePurposeFromClient(
+      purposeId: PurposeId,
+      clientId: ClientId,
+      logger: Logger,
+      correlationId: string
+    ) {
+      const token = (await refreshableToken.get()).serialized;
+      const headers = getHeaders(correlationId, token);
+      await authMgmtClient.removeClientPurpose(undefined, {
+        params: { purposeId, clientId },
+        withCredentials: true,
+        headers,
+      });
+      logger.info(`Deleting purpose ${purposeId} from client ${clientId}`);
+    },
+    async updatePurposeState(
+      purposeId: PurposeId,
+      versionId: PurposeVersionId,
+      state: ApiClientComponentState,
+      logger: Logger,
+      correlationId: string
+    ) {
+      const token = (await refreshableToken.get()).serialized;
+      const headers = getHeaders(correlationId, token);
+      await authMgmtClient.updatePurposeState(
+        { versionId, state },
+        {
+          params: { purposeId },
+          withCredentials: true,
+          headers,
+        }
+      );
+      logger.info(`Updateding Puprpose ${purposeId} state for all clients`);
```

Comment:
Remember to fix the typos before merging

Reply:
Puprpose 😅